internal/pkg/keychain/keychain_darwin.go

```diff
@@ -0,0 +1,98 @@
+//go:build darwin
+// +build darwin
+
+package keychain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/netrc"
+	"github.com/keybase/go-keychain"
+)
+
+const (
+	accessGroup = "cli"
+	isCloud     = true
+	sep         = "?"
+)
+
+func WriteKeychainItem(ctxName, url, username, password string) error { // this happens after login, so you can have the ctxName
+	service := netrc.GetLocalCredentialName(isCloud, ctxName)
+
+	item := keychain.NewGenericPassword(service, url, username+"-"+url, []byte(username+sep+password), accessGroup) // save username as the label
+	item.SetSynchronizable(keychain.SynchronizableNo)
+	item.SetAccessible(keychain.AccessibleWhenUnlocked)
+
+	err := keychain.AddItem(item)
+	if err == keychain.ErrorDuplicateItem {
```

Comment:
What if there's an error that's not "duplicate item"?

Reply:
it'll return the err on the next line

Reply:
But it'll print "saved item using: ...". Did you mean to delete that line?

Reply:
yeah all println is only for my testing and reviewing.. they'll be deleted